include/openthread/netdata.h

```diff
@@ -230,6 +230,19 @@ struct otJoinerDiscerner;
 otError otNetDataSteeringDataCheckJoinerWithDiscerner(otInstance *                    aInstance,
                                                       const struct otJoinerDiscerner *aDiscerner);
 
+/**
+ * This function checks whether the Leader's Network Data contains a given OMR prefix.
+ *
+ * @param[in]  aInstance  A pointer to an OpenThread instance.
+ * @param[in]  aPrefix    A pointer to the IPv6 prefix.
+ *
+ * @returns  Whether Leader's Network Data contains the OMR prefix @p aPrefix.
+ *
+ * @note This API is only available when `OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE` is used.
```

Comment:
It has to for now since both `IsValidOmrPrefix` methods are in `RoutingManger`. 

Refactoring `RoutingManger::IsValidOmrPrefix` methods would be well beyond the scope of this PR. 

We may remove `OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE` restrictions in a future PR that refactors IsValidOmrPrefix.

Thoughts?

Reply:
Will be refactored in a future PR.